cmd/helpers/doc.go

```diff
@@ -0,0 +1,53 @@
+package helpers
+
+/*
+Helpers package abstracts the following from the application that uses these
+helpers (using dependency injection):
+
+ - error handling for all non-control flow scenarios
+ - trace support
+
+The above abstractions enable the application code (using these helpers) to not
+have to sprinkle if (err != nill) blocks (except when the application wants to
+affect application flow based an err)
+
+Do and Do Not:
+ - Do verify parameter values and panic if the helper function would be unable
+   to succeed, to catch coding errors (do not panic for user input errors)
+ - Do not output (except for in the `output` helper of course). Do use the injected
+    trace method to output low level debugging information
+ - Do not return error if client is not going use the error for control flow, call the
+   injected checkErr instead, which will probably end up calling cobra.checkErr and exit:
+     e.g. Do not sprinkle application (non-helper) code with:
+       err, _ := fmt.printf("Hope this works")
+       if (err != nil) {
+         panic("How unlikely")
+       }
+     Do use the injected checkErr callback and let the application decide what to do
+       err, _ := printf("Hope this works)
```

Comment:
> err, _ := printf("Hope this works) [](http://example.com/codeflow?start=6&length=35)

is there a class of functions that return error where it's ok to ignore the return completely? 
So far for pkg/sqlcmd and cmd/sqlcmd I've just ignored all errors from stdout etc.

Reply:
At the moment, if it returns err I'm checking it with a checkErr(err) call, and letting the injected handler make a decision on whether to ignore it.  If we get a non nil err I think I want to know about it for now, let's find out what unexpected things fail.